src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java

```diff
@@ -147,10 +145,25 @@
     }
 
     private static Optional<Set<Reference<Product>>> getProductReferenceSet(@Nonnull final Attribute attribute) {
-        try {
-            return Optional.of(attribute.getValue(AttributeAccess.ofProductReferenceSet()));
-        } catch (final JsonException exception) {
-            return Optional.empty();
+        return isProductReferenceSet(attribute)
+            ? Optional.of(attribute.getValue(AttributeAccess.ofProductReferenceSet())) : Optional.empty();
+    }
+
+    static boolean isProductReference(@Nonnull final Attribute attribute) {
+        final JsonNode valueAsJsonNode = attribute.getValueAsJsonNode();
+        return !(valueAsJsonNode instanceof ArrayNode) && isValueAProductReference(valueAsJsonNode);
+    }
+
+    static boolean isProductReferenceSet(@Nonnull final Attribute attribute) {
+        final JsonNode valueAsJsonNode = attribute.getValueAsJsonNode();
+        return (valueAsJsonNode instanceof ArrayNode) && isValueAProductReference(valueAsJsonNode.elements().next());
+    }
+
+    private static boolean isValueAProductReference(@Nonnull final JsonNode valueAsJsonNode) {
+        if (valueAsJsonNode.isContainerNode()) {
+            final JsonNode typeIdNode = valueAsJsonNode.get("typeId");
+            return typeIdNode != null && Objects.equals(typeIdNode.asText(), "product");
```

Comment:
No need to use `Objects.equals()` in this case: `"product".equals(typeIdNode.asText())` is sufficient enough since `"product"` is definitely not null

Reply:
bec3c5d